lib/improver/utilities/temporal_interpolation.py

```diff
@@ -149,76 +149,51 @@ def construct_time_list(self, initial_time, final_time):
         return [('time', time_list)]
 
     @staticmethod
-    def check_cube_coords_dtype(cube, new_cube, dim_coords=None):
+    def enforce_time_coords_dtype(cube):
         """
-        Update the data type of the coords within the new_cube
-        to those in the original cube. This can be limited to a list of
-        dimension coordinates and associated auxillary coordinates if required.
+        Enforce the data type of the time, forecast_reference_time and
+        forecast_period within the cube, so that time coordinates do not
+        become mis-represented. The units of the time and
+        forecast_reference_time are enforced to be
+        "seconds since 1970-01-01 00:00:00" with a datatype of int64.
+        The units of forecast_period are enforced to be seconds with a datatype
+        of int32.
```

Comment:
Function modifies cube coordinates in-place - worth mentioning in docstring.

Reply:
Added to docstring.